File: Gauss_Jacobi.swift
//Matrix Declearation 
var mat = [[Double]]()
var n: Int
print("Enter the size of the matrix :")
n = Int(readLine()!)!
print("Enter the values of matrix : \n")
for _ in 0..<n{
var row = [Double]()
for _ in 0..<n{
row.append(Double(readLine()!)!)
}
mat.append(row)
}

//b vector
print("Enter the b vector elements :")
var b=[Double]()
for _ in 0..<n{
b.append(Double(readLine()!)!)
}

//x vector
print("Enter the initial values of x vector :")
var x=[Double]()
for _ in 0..<n{
x.append(Double(readLine()!)!)
}

print("mat = \(mat) \n")
print("b = \(b) \n")
print("x = \(x) \n")

var itr: Int
print("Enter the no. of iterations : ")
itr = Int(readLine()!)!

print("Enter the dummy vector values same as x vector initial value :")
var xn=[Double]()
for _ in 0..<n{
//xn[p] = x[p]
xn.append(Double(readLine()!)!)
}

for _ in 0..<itr{
for e in 0..<n{
for f in 0..<n{

if e < f {
x[e]=(b[e]-mat[e][f]*x[f])/mat[e][e]
}

if e > f {
x[e]=(b[e]-mat[e][f]*xn[f])/mat[e][e]
}

}
}

for q in 0..<n{
xn[q] = x[q]//xn.append(x[q])
}

}

print("Results using Gauss Jacobi : ")
for v in 0..<n{
print("x[\(v)] = \(x[v])")
}

File: Gauss_Jacobi.cs
/******************************************************************************

                            Online C# Compiler.
                Code, Compile, Run and Debug C# program online.
Write your code in this editor and press "Run" button to execute it.

*******************************************************************************/

using System;
class GaussJacobi {
  static void Main() {
    Console.WriteLine("Enter the size of the matrix : ");
    int n;
    n = Convert.ToInt32(Console.ReadLine());
    int[,] mat = new int[n,n];
    Console.WriteLine("Entert the values : ");
    
    for(int i=0;i<n;i++){
        
        for(int j=0;j<n;j++){
            mat[i,j] = Convert.ToInt32(Console.ReadLine());
        }
    }
    Console.WriteLine("Matrix is : ");
    for(int i=0;i<n;i++){
        
        for(int j=0;j<n;j++){
             Console.WriteLine(mat[i,j]);//mat[i,j] = Convert.ToInt32(Console.ReadLine());
        }
    }
   //int[] b;
   Console.WriteLine("Enter the b vector : ");
    int[] b = new int[n];
    for(int i=0;i<n;i++){
     b[i] =  Convert.ToInt32(Console.ReadLine());   
    }
    
    Console.WriteLine("Enter the initial values of x vector : ");
    float[] x = new float[n];
    for(int i=0;i<n;i++){
     x[i] =  float.Parse(Console.ReadLine());   
    }
    
    //Dummy vector 
    float[] xn = new float[n];
    for(int i=0;i<n;i++){
     xn[i] =  x[i];   
    }
    int itr;
    Console.WriteLine("Enter the no. of iterations : ");
    itr = int.Parse(Console.ReadLine());
    for(int k=0;k<itr;k++){
    for(int i=0;i<n;i++){
        for(int j=0;j<n;j++){
            if(i<j){
               x[i] = (b[i]-mat[i,j]*x[j])/mat[i,i]; 
            }
            if(i>j){
                x[i] = (b[i]-mat[i,j]*xn[j])/mat[i,i];
            }
        }
      }
      for(int i=0;i<n;i++){
     xn[i] =  x[i];   
     }
    
    }
    Console.WriteLine("Results using Gauss-jacobi : ");
    Console.WriteLine("x[0] = "+x[0]+","+x[1] = "+x[1]);
    
    
  }
}
